src/components/VisTypeIcon.js

```diff
@@ -0,0 +1,162 @@
+import {
+    IconTable16,
+    IconVisualizationArea16,
+    IconVisualizationAreaStacked16,
+    IconVisualizationBar16,
+    IconVisualizationBarStacked16,
+    IconVisualizationColumn16,
+    IconVisualizationColumnMulti16,
+    IconVisualizationColumnStacked16,
+    IconVisualizationGauge16,
+    IconVisualizationLine16,
+    IconVisualizationLineMulti16,
+    IconVisualizationPie16,
+    IconVisualizationRadar16,
+    IconVisualizationScatter16,
+    IconVisualizationSingleValue16,
+    IconTable24,
+    IconVisualizationArea24,
+    IconVisualizationAreaStacked24,
+    IconVisualizationBar24,
+    IconVisualizationBarStacked24,
+    IconVisualizationColumn24,
+    IconVisualizationColumnMulti24,
+    IconVisualizationColumnStacked24,
+    IconVisualizationGauge24,
+    IconVisualizationLine24,
+    IconVisualizationLineMulti24,
+    IconVisualizationPie24,
+    IconVisualizationRadar24,
+    IconVisualizationScatter24,
+    IconVisualizationSingleValue24,
+} from '@dhis2/ui'
+import PropTypes from 'prop-types'
+import React from 'react'
+import {
+    VIS_TYPE_PIVOT_TABLE,
+    VIS_TYPE_COLUMN,
+    VIS_TYPE_STACKED_COLUMN,
+    VIS_TYPE_BAR,
+    VIS_TYPE_STACKED_BAR,
+    VIS_TYPE_LINE,
+    VIS_TYPE_AREA,
+    VIS_TYPE_STACKED_AREA,
+    VIS_TYPE_PIE,
+    VIS_TYPE_RADAR,
+    VIS_TYPE_GAUGE,
+    VIS_TYPE_YEAR_OVER_YEAR_LINE,
+    VIS_TYPE_YEAR_OVER_YEAR_COLUMN,
+    VIS_TYPE_SINGLE_VALUE,
+    VIS_TYPE_SCATTER,
+} from '../modules/visTypes'
+
+export const VisTypeIcon = ({ type, size = 24, ...props }) => {
+    let VisIcon
+
+    switch (type) {
+        case VIS_TYPE_PIVOT_TABLE: {
+            VisIcon = size === 16 ? IconTable16 : IconTable24
```

Comment:
Not sure about this pattern of using a number as an enum. Looks like it would be better to use a const like `LARGE_ICON` or something, since all values except specifically 16 and 24 will work 🤔

Reply:
or if there will ever only be 2 sizes, maybe have a prop called ´useSmall´ that just switches to the alternative size?

Reply:
So far it's only 2 sizes that `ui` provides, so perhaps the latter solution is better.

Reply:
Refactored to use `useSmall`.